pynamodb/asyncio/batch_get.py

```diff
@@ -0,0 +1,62 @@
+import asyncio
+import typing
+
+from typing import List, Dict, Any, TypeVar
+
+from aioitertools.asyncio import as_completed
+from more_itertools import chunked
+
+from pynamodb.constants import BATCH_GET_PAGE_LIMIT
+
+if typing.TYPE_CHECKING:
+    from pynamodb.models import Model # noqa: F401
+
+_T = TypeVar("_T", bound="Model")
+
+class BatchGetIterator(typing.AsyncIterator[_T]):
+
+    def __init__(self, model_cls: typing.Type[_T], keys: typing.Iterable, consistent_read: bool | None = None, attributes_to_get: typing.Sequence[str] | None = None):
+        self.model_cls = model_cls
+        self.unprocessed_batch_items = list(keys)
+        self.consistent_read = consistent_read
+        self.attributes_to_get = attributes_to_get
+        self.current_batch: List[Dict[str, Any]] = []
```

Comment:
nitpick, but the new documented preferred way to type is using `list` and `dict` directly rather than `typing.List` and `typing.Dict` - I highly doubt there will ever be any real impact but it does mean fewer imports, which I like